dts/arm/atmel/samd.dtsi

```diff
@@ -26,6 +26,116 @@
 		reg = <0x20000000 0x8000>;
 	};
 
+	clocks {
+		#address-cells = <0>;
+
+		osc32k: osc32k {
+			compatible = "fixed-clock";
+			clock-frequency = <32768>;
+			#clock-cells = <0>;
+			status = "disabled";
+			label = "OSC32K";
+		};
+
+		ulposc32k: ulposc32k {
+			compatible = "fixed-clock";
+			clock-frequency = <32768>;
+			#clock-cells = <0>;
+			label = "ULPOSC32K";
+		};
+
+		osc8m: osc8m {
+			compatible = "fixed-clock";
+			clock-frequency = <8000000>;
+			#clock-cells = <0>;
+			label = "OSC8M";
+		};
+
+		xosc32k: xosc32k {
+			compatible = "fixed-clock";
+			clock-frequency = <32768>;
+			#clock-cells = <0>;
+			status = "disabled";
+			label = "XOSC32K";
+		};
+
+		xosc: xosc {
```

Comment:
samd5x also has two of these, calling it `xosc0` should make sharing code easier.